### PR TITLE
NOTIF-450 Move tests to the engine module

### PR DESCRIPTION
**Pull request by gwenneg**
This PR moves most of the `engine` tests to the `engine` module. It does not move `LifecycleITest` yet because this class will require extra attention in a dedicated PR.

### CODE REVIEW

File: engine/src/test/java/com/redhat/cloud/notifications/db/ResourceHelpers.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class ResourceHelpers {
+
+    @Inject
+    EmailAggregationRepository emailAggregationRepository;
+
+    public Uni<Boolean> addEmailAggregation(String accountId, String bundleName, String applicationName, JsonObject payload) {
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/db/ResourceHelpers.java:17]: Moved from `backend`, code unchanged.

File: engine/src/test/java/com/redhat/cloud/notifications/TestLifecycleManager.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.redhat.cloud.notifications.events.EventConsumer.INGRESS_CHANNEL;
+import static com.redhat.cloud.notifications.events.FromCamelHistoryFiller.EGRESS_CHANNEL;
+import static com.redhat.cloud.notifications.events.FromCamelHistoryFiller.FROMCAMEL_CHANNEL;
+import static com.redhat.cloud.notifications.processors.camel.CamelTypeProcessor.TOCAMEL_CHANNEL;
+import static com.redhat.cloud.notifications.processors.email.EmailSubscriptionTypeProcessor.AGGREGATION_CHANNEL;
+
+public class TestLifecycleManager implements QuarkusTestResourceLifecycleManager {
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/TestLifecycleManager.java:23]: Copied from `backend/.../TestLifecycleManager` with only 2 lines of code changed (properties).

File: engine/src/test/java/com/redhat/cloud/notifications/MicrometerAssertionHelper.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * {@link MeterRegistry#clear()} cannot be called between tests to reset counters because it would remove the counters
+ * instances from the registry while we use many {@link ApplicationScoped} beans which hold references pointing at the
+ * aforementioned counters instances. That's why we need this helper to assert increments in a reliable way.
+ */
+@ApplicationScoped
+public class MicrometerAssertionHelper {
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/MicrometerAssertionHelper.java:23]: Copied from `backend/.../MicrometerAssertionHelper`, code unchanged. The `backend` class will be completely removed eventually, but not in this PR.

File: common/src/test/java/com/redhat/cloud/notifications/ReflectionHelper.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.redhat.cloud.notifications;
+
+import java.lang.reflect.Field;
+
+public class ReflectionHelper {
+
+    public static <T> void updateField(T object, String field, Object value, Class<T> klass) {
```
Comment (gwenneg) [inline common/src/test/java/com/redhat/cloud/notifications/ReflectionHelper.java:7]: Moved from `backend/.../TestHelpers`, code unchanged.

File: backend/src/test/java/com/redhat/cloud/notifications/MockServerClientConfig.java [hunk 5]
```diff
@@ -74,4 +79,13 @@ public String getRunningAddress() {
         return String.format("%s:%d", mockServerClient.remoteAddress().getHostName(), mockServerClient.remoteAddress().getPort());
     }
 
+    public static String getFileAsString(String filename) {
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/MockServerClientConfig.java:82]: Moved from `backend/.../TestHelpers`, code unchanged.

File: backend/src/test/java/com/redhat/cloud/notifications/MockServerClientConfig.java [hunk 6]
```diff
@@ -74,4 +79,13 @@ public String getRunningAddress() {
         return String.format("%s:%d", mockServerClient.remoteAddress().getHostName(), mockServerClient.remoteAddress().getPort());
     }
 
+    private static String getFileAsString(String filename) {
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/MockServerClientConfig.java:82]: Moved from `backend/.../TestHelpers`, code unchanged.

File: engine/src/test/java/com/redhat/cloud/notifications/TestHelpers.java [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+
+import java.time.LocalDateTime;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+import static com.redhat.cloud.notifications.Constants.X_RH_IDENTITY_HEADER;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class TestHelpers {
```
Comment (gwenneg) [inline engine/src/test/java/com/redhat/cloud/notifications/TestHelpers.java:20]: Copied/moved from `backend/.../TestHelpers`.

**Comment by gwenneg** [comment]
@josejulio This is still a draft because I'm working on the tests stability. Sometimes Quarkus fails to start because of a NPE and the tests won't run. But you can already have a look at this PR as I don't think the code will change now.

**Comment by gwenneg** [comment]
Tests are stable now.

**Comment by gwenneg** [comment]
The CodeQL failure is unrelated (JS).

**Review by josejulio — APPROVED** [review]

**Pull request merged**